[CS2113-F10-4] AlloNUS

wli-linda opened the pull request:
AlloNUS (“all on us”) is an all-in-one platform for tracking your classes, expenses, and personal contacts optimized for use via a Command Line Interface (CLI). If you can type fast, AlloNUS can get your schedule, expenses and contact management tasks done faster than traditional GUI apps.

wli-linda committed:
Merge pull request #43 from wli-linda/22-contacts-enhancements

Add more fields to Contact

YitHien committed:
Minor changes to User Guide for Expense Tracker

OzairHasan committed:
Change logger log levels and build jar file

OzairHasan committed:
Merge pull request #45 from OzairHasan/master

Change logger log levels and build jar file

wli-linda committed:
Update contacts manager v1.0 features in UG

wli-linda committed:
Implement naive find feature for contacts

Currently only looks for matching "name" fields

wli-linda committed:
Implement naive edit feature for contacts

Only supports updating default field (name)

wli-linda committed:
Merge pull request #51 from wli-linda/contacts-v2.0-features

Add rough implementation of Contacts Manager v2.0 features

wli-linda committed:
Update headers for DG

wli-linda committed:
Merge pull request #54 from wli-linda/contacts-v2.0-features

Update headers for DG

YitHien committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-4/tp

wli-linda committed:
Make Contact class more OOP

Contact class fields are all using String type.

Making the fields to be separate types that extend from the Field type is more OOP, and allows for some polymorphism that may be helpful when parsing.

Let's:
* Create new classes for each Contact field
* Update existing code to ensure compatibility

wli-linda committed:
Support editing multiple contact fields

`editContact()` only edits the name of a `Contact` object.

A user may want to edit more than the name field, and it would be more flexible if they can edit multiple fields at the same time.

Let's:
* refactor `parseContact` such that there is a separate method extracted for setting contact fields supplied by the user
* call the abovementioned extracted method in `editContact()` to set the fields

mathanmahe committed:
Add basic find and edit features for Module Manager

mathanmahe committed:
Match gradle checkstyle

YitHien committed:
Add skeleton functionality for Edit feature

YitHien committed:
Fix typo in text ui checks

YitHien committed:
Add Find feature for Expense Tracker, update user guide and add more Junit tests

YitHien committed:
Refactor methods out of long ExpenseRunner method for better SLAP

YitHien committed:
Minor changes to fit Gradle checkstyle

YitHien committed:
Add minor changes to String constants

wli-linda committed:
Improve JUnit Test coverage for contacts

mathanmahe committed:
Add minimal design details in DG

wli-linda committed:
Include PlantUML diagrams in DG for contacts

mathanmahe committed:
Improve code quality

mathanmahe committed:
Refactor remaining String objects as constants

mathanmahe committed:
Change access modifiers to private for all constants

YitHien committed:
Add draft Developer Guide for Expense Tracker

OzairHasan committed:
Update DG for menu features and include UML diagrams

OzairHasan committed:
Update UG for main menu features

OzairHasan committed:
Update to agree with gradle checks

mathanmahe committed:
Update DG with manual testing instructions

wli-linda committed:
Merge pull request #56 from wli-linda/master

Make Contacts Manager features more robust

mathanmahe committed:
Merge pull request #64 from mathanmahe/modules-edit-find

Add basic find and edit features for Module Manager

YitHien committed:
Merge pull request #65 from YitHien/YitHien-EditSkeleton

OzairHasan committed:
Merge branch 'master' into menu-DG-update

OzairHasan committed:
Merge pull request #71 from OzairHasan/menu-DG-update

Menu DG and UG update

YitHien committed:
Minor bug fix

YitHien committed:
Merge pull request #72 from YitHien/master

Minor bug fix to Expense Tracker Junit Test

YitHien committed:
Minor changes to file names

YitHien committed:
Merge branch 'master' into YitHien-DeveloperGuideDraft

YitHien committed:
Merge pull request #69 from YitHien/YitHien-DeveloperGuideDraft

Add Draft Developer Guide for Expense Tracker

mathanmahe committed:
Improve code quality by refactoring recurring strings as constants

mathanmahe committed:
Update User Guide for edit module feature

mathanmahe committed:
Fix bug in find modules feature that allowed whitespace find query

mathanmahe committed:
Add new exception to classify invalid find inputs

mathanmahe committed:
Update find feature in UG

mathanmahe committed:
Improve find and edit code quality

mathanmahe committed:
Edit build.gradle to include ical4j and sl4j nop dependencies for ics parsing

mathanmahe committed:
Add ical4j jar for compile time

mathanmahe committed:
Implement skeleton for parsing .ics calendar files

mathanmahe committed:
Integrate ModuleCalendarReader with StudyManager class

mathanmahe committed:
Match gradle checkstyle test

mathanmahe committed:
Improve code quality of read ics feature

YitHien committed:
Add constraints on Date and Amount fields, add more custom exceptions

YitHien committed:
Add constraints on Date and Amount fields, add more custom exceptions

YitHien committed:
Fix bug for config file

YitHien committed:
Attempt bug fix for Gradle

mathanmahe committed:
Add minor fix

mathanmahe committed:
Rebuild after gradle error

YitHien committed:
Minor change to Javadoc

OzairHasan committed:
Add load-save functionality

wli-linda committed:
Add JavaDoc comments in contacts package

wli-linda committed:
Add v2.0 contact features in UG

wli-linda committed:
Make find contact more robust and add tests

wli-linda committed:
Update DG for contacts with class diagram

YitHien committed:
Merge pull request #75 from YitHien/YitHien-ParseDateField

Add constraints on Date and Amount fields; add additional custom exceptions and Junit tests

mathanmahe committed:
Merge pull request #76 from mathanmahe/modules-enhancements

Add enhancements to find, edit features and add parse ics feature

OzairHasan committed:
Merge branch 'master' into load-save-feature

OzairHasan committed:
Merge pull request #77 from OzairHasan/load-save-feature

Add load-save functionality

wli-linda committed:
Merge pull request #78 from wli-linda/master

Improve Contacts Manager's related documentation

YitHien committed:
Add minor changes to UserGuide; bug fix for Date formatting

YitHien committed:
Bug fix in ExpenseParserTest

YitHien committed:
Bug fix for text UI testing

OzairHasan committed:
Merge pull request #81 from YitHien/YitHien-UGUpdate

Add minor changes to UserGuide

mathanmahe committed:
Update DG with StudyManager implementation and user stories.

mathanmahe committed:
Update UserGuide to include read ics feature of StudyManager

mathanmahe committed:
Add improvement to command summary

mathanmahe committed:
Fix bug that allowed empty edit parameters

mathanmahe committed:
Update text-ui test for edit and find

mathanmahe committed:
Introduce new valid field "Laboratory" for module category

mathanmahe committed:
Add text-ui test case for read ics feature of StudyManager

mathanmahe committed:
Reflect the change in StudyManager logic in StorageFile

mathanmahe committed:
Clear text file that was causing error

mathanmahe committed:
Remove test cases for parse ics that caused previous error

mathanmahe committed:
Remove minor test case

mathanmahe committed:
Implement workaround over compatibility issue for parse ics testing

OzairHasan committed:
Refine save load functionality and add jUnit test for StorageFile class

OzairHasan committed:
Update DG and UG for load-save functionality

OzairHasan committed:
Update JUnit test for StorageFile

alanlowzies commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,27 @@
 
```
Perhaps the Developer Guide title could include the application's name (e.g. Developer Guide for AlloNUS) so that the reader will know that he/she is reading the right document.

Anvitha-r replied:
could start the developers guide by having a contents page to indicate the components in the Developers guide

alanlowzies commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,27 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Study Manager component
+API: `StudyManager.java`
+
+![](images/StudyManagerClass.png)
+
+The `StudyManager` component,
+1. Stores the academic schedule, i.e. all `Module` objects are contained in a `modulesList` object.
+2. The storage can be bound to `modulesList` list such that everytime a change is observed in the list it is saved on to a file.
+3. Does not depend on other components like `ExpenseTracker` and `ContactsManager`
```
Is it possible for the markup colour to be changed to a brighter colour? I noticed that the markups on the GitHub Page of your documentation kind of blend into the dark background and can be potentially difficult for readers to read.

Here's a screenshot:
![image](https://user-images.githubusercontent.com/30099983/160956224-562ccc5a-5a9d-4750-b52e-503829a82aa3.png)

alanlowzies commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,27 @@
+
+
 
 ## Product scope
 ### Target user profile
```
Is the 'Product Scope' section wrongly positioned? I noticed that currently, the 'Main Menu' section is a subheader of 'Product Scope', and I presume that the former should be under the 'Design and Implementation' section.

Anvitha-r replied:
can include information relevant to target user profile, the project scope

alanlowzies commented on docs/DeveloperGuide.md:
```diff
@@ -25,6 +40,159 @@
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
 
+### Main Menu
```
Perhaps the capitalisation of the first letter of the headings' words can be standardised? I noticed that some headings only have the first word's letter capitalised (e.g. 'Design & implementation') while other headings capitalise every other word's first letter (e.g. 'Main Menu').

alanlowzies commented on docs/DeveloperGuide.md:
```diff
@@ -25,6 +40,159 @@
+
+**API:** `AllOnUs.java`
+
+The menu is accessed through a call to static method main() of the AllOnUs class. 
+The Class Diagram below shows the overall structure of the application from the menu's point of view.
+The AllOnUs class and other classes that are coupled to the menu are therefore included.
```
Perhaps method calls and class names can be marked up? For example, on the first line of this highlighted area:
> "... call to static method `main()` of the `AllOnUs` class. ..."

alanlowzies commented on docs/DeveloperGuide.md:
```diff
@@ -25,6 +40,159 @@
+The Class Diagram below shows the overall structure of the application from the menu's point of view.
+The AllOnUs class and other classes that are coupled to the menu are therefore included.
+
+![](images/MenuFeaturesClassDiagram.png)
```
Nice class diagram! Clear, concise, and complements the write up of the class. Details of other classes that are not necessarily important to this section are also omitted for better readability.

alanlowzies commented on docs/DeveloperGuide.md:
```diff
@@ -25,6 +40,159 @@
+
+![](images/MenuFeaturesSequenceSubDiagram.png)
+
+As illustrated in the diagram, once main() method of AllOnUs is called, a new unnamed :AllOnUs object is created 
+for which we execute the non-static method run(), which contains a loop for sustained interaction with the user
+(command line). Objects for the ContactsManager (contactsManager), StudyManager (studyManager) and ExpenseTracker (expenseTracker)
+classes are also created, where Logger (logger) and TextUi (ui) objects are already contained in the fields of the 
+:AllOnUs object. 
```
Are the object creations of `AllOnUs`, `ContactsManager`, `StudyManager`, and `ExpenseTracker` not properly represented in the sequence diagram? I notice that the highlighted paragraph mentions these classes being instantiated, but the sequence diagram does not really show any objects being created.

Here's a screenshot of the sequence diagram that I was referring to:

![image](https://user-images.githubusercontent.com/30099983/160957935-30b0ee4a-feee-4b66-a5c6-e70619c43698.png)

My sincere apologies if I misunderstood your sequence diagram and phrasing 🙏🏻

alanlowzies commented on docs/DeveloperGuide.md:
```diff
@@ -25,6 +40,159 @@
+command. 
+
+Command to enter a particular section: 
+The returned string from getUserInput() call becomes a parameter for a function call to static method is*Command(), which
+belongs to the AllOnUs class, and the is*Command() method is of the form isStudyManagerCommand(), isContactsManagerCommand(),
```
Perhaps the asterisk in `is*Command()` can be escaped? I noticed that the asterisks are being interpreted as italics.

Screenshot of example:
![image](https://user-images.githubusercontent.com/30099983/160958883-8d504c23-cf27-45f0-9f2a-40f0254a7cf3.png)


Alternatively, you could mark up the `is*Command()` methods :)

alanlowzies commented on docs/DeveloperGuide.md:
```diff
@@ -25,6 +40,159 @@
+
+The following sequence diagram shows how the budget is checked upon every new addition of expense records:
+
+![](images/BudgetingReminderSequenceDiagram.png)
```
For this diagram, perhaps the return notation of `budget(int)` can be rephrased to `budget:int`? The current notation may be misunderstood as a method call.

![image](https://user-images.githubusercontent.com/30099983/160960005-745f3eec-2975-4375-98d6-920eeba58e0b.png)

Anvitha-r replied:
the budget is it calling int or its it of type int, maybe a bit confusing, could use appropriate notations; budget : Int

Anvitha-r reviewed (commented):
![Screenshot 2022-03-31 at 9 49 08 AM](https://user-images.githubusercontent.com/69703144/160960216-cfc1882c-068c-44fa-9e49-c9c7607ac89a.png)
![Screenshot 2022-03-31 at 9 48 12 AM](https://user-images.githubusercontent.com/69703144/160960224-65492d3b-3416-4100-9cad-090b17e566ab.png)
![Screenshot 2022-03-31 at 9 48 06 AM](https://user-images.githubusercontent.com/69703144/160960241-3c4de186-1329-4159-8cee-75cd805445b0.png)

alanlowzies reviewed (commented):
Good job with the developer guide! I have left some suggestions for your consideration. Thank you! :)

Anvitha-r commented on docs/DeveloperGuide.md:
```diff
@@ -25,6 +40,159 @@
+Note: Exception classes are left out of this diagram that aims to show the core structure of the application. is*Command()
+method here refers to all methods of this form (more on this will be elaborated later in detail). 
+
+As illustrated in the Class Diagram above, the AllOnUs class only associates to one of each of the other classes.
```
the explanations maybe a bit too long for the reader perhaps could shorten the explanation

Anvitha-r reviewed (commented):
![Screenshot 2022-03-31 at 9 56 14 AM](https://user-images.githubusercontent.com/69703144/160960916-ddef710c-6a9e-46ec-b580-e42f5a07b729.png)

Anvitha-r reviewed (commented):
![Screenshot 2022-03-31 at 10 06 42 AM](https://user-images.githubusercontent.com/69703144/160962004-e2d745e9-0d6b-4999-bc5d-12509e1bf1e2.png)

Anvitha-r commented on docs/DeveloperGuide.md:
```diff
@@ -25,6 +40,159 @@
+command. 
+
+Command to enter a particular section: 
+The returned string from getUserInput() call becomes a parameter for a function call to static method is*Command(), which
```
perhaps the asterix is being recognised as a formatting style instead, could ensure that this is taken care of to minimize confusion.

Anvitha-r reviewed (commented):
![Screenshot 2022-03-31 at 10 15 59 AM](https://user-images.githubusercontent.com/69703144/160962912-977083f1-64ff-4b3a-b297-9a3c46e2da1a.png)

mathanmahe committed:
Change library to read timezones that was compatibility issues

mathanmahe committed:
Make file reading OS independent

mathanmahe committed:
Remove minor test case

OzairHasan committed:
Attempt to resolve gradle issue

OzairHasan committed:
Attempt to debug gradle error

OzairHasan committed:
Table changes to StorageFileTest.java for now

OzairHasan committed:
Remove debug lines

mathanmahe committed:
Merge pull request #82 from mathanmahe/modules-v2.0-documentation

Update documentation and tests for v2.0 release

OzairHasan committed:
Merge pull request #88 from OzairHasan/load-save-refinement-DG

Load and save refinement and DG/UG/JUnit update

mathanmahe committed:
Fix bug that showed warning messages on restart

mathanmahe committed:
Merge pull request #93 from mathanmahe/bug-fix

Fix bug that showed warning messages on restart

YitHien committed:
Minor typo in UserGuide

YitHien committed:
Merge pull request #94 from YitHien/master

Minor typo in UserGuide  - Expense Tracker List

johnsuharjono commented on docs/DeveloperGuide.md:
```diff
@@ -2,29 +2,310 @@
 
```
Perhaps adding a table of content with hyperlink to each of the section would be helpful!

cristoforows commented on docs/DeveloperGuide.md:
```diff
@@ -2,29 +2,310 @@
+will carry out what it needs to and then eventually return to the run() method of the :AllOnUs object. If false is returned, we proceed 
+to the next check on the string. This is also how the other commands and methods of the same form work. 
+
+Command for help section:
+If the isHelpCommand() method returns true on the string representing user input, the displayHelp() static method is called (in AllOnUs class)
+which displays the help section to the user and then returns to the run() method of :AllOnUs object.
+
+Empty of unrecognized command:
+When the returned string is empty, we simply continue to the next iteration of the loop. This is essentially similar to a "pass" statement
+as when the user does not input anything, the application silently waits for the user to input something into the command line. 
+When there is an unrecognized command, the application prompts the user through a function call to a static method printInvalidMainMenuCommandMessage()
+(belongs to AllOnUs class and not included in the diagrams above) to enter a known command and suggests referring to the help section.
+
+Exit command:
+If the isExitCommand() method returns true on the string representing user input, the loop breaks, and control is returned to the static main()
+method of the AllOnUs class, which then calls the static exit method in the same class to print a termination message, and then finally control
+is returned to the OS. 
```
Would it be better to make it as bullet point to ensure readability

johnsuharjono replied:
Perhaps can use markup on method calls like `isExitCommand()` instead of normal text to improve readability

johnsuharjono commented on docs/DeveloperGuide.md:
```diff
@@ -2,29 +2,310 @@
+These classes are, namely, the ContactsManager class which manages the contacts section of the application, 
+the StudyManager class which manages the module and timetable part of the application, the ExpenseTracker class
+which manages the expenses section of the application, the Logger class which helps with logging information about the 
+state of the programme when it is running and the TextUi class which handles user input from command line.
```
Perhaps rather than using paragraph would be better if list of class were use to represent as it is simpler and more streamlined

johnsuharjono commented on docs/DeveloperGuide.md:
```diff
@@ -2,29 +2,310 @@
+
+Step 1: The user navigates to the Expense Tracker for the first time, and decides to set a budget of $300.
 
+![](images/BudgetingReminder0.png)
+
+Step 2: The user proceeds to add 2 expense records, each with an amount of $100.
+
+![](images/BudgetingReminder1.png)
+
+Step 3: The user now tries to add 1 more expense record with an amount of $200. This exceeds the budget and hence the 
+addition will not be processed.
+
+![](images/BudgetingReminder2.png)
+
+The following sequence diagram shows how the budget is checked upon every new addition of expense records:
+
+![](images/BudgetingReminderSequenceDiagram.png)
```
Perhaps the notation for `budget(int)` change to `budget:int` could be fix as to avoid misunderstanding.

xseh replied:
The activation bar should end on the final return value.

cristoforows commented on docs/DeveloperGuide.md:
```diff
@@ -2,29 +2,310 @@
+`parseXYZExpense` where `XYZ` is a placeholder for the specific command name (eg. `parseDeleteExpense`).
+4. After parsing, the command is executed accordingly within the `ExpenseTracker` class.
+
+#### [Proposed] Budgeting Reminder Feature
```
perhaps can make the header a bit bigger?
![image](https://user-images.githubusercontent.com/75602488/161185914-a2605b16-2bc2-4729-9d6f-70d923f9c735.png)

johnsuharjono commented on docs/DeveloperGuide.md:
```diff
@@ -2,29 +2,310 @@
+|v1.0| user|delete an expense|remove expenses that I do not need to track|
+|v1.0| user|add a new contact|keep track of my contacts|
+|v1.0| user|list all contacts|get a list of all my contacts in one place|
+|v1.0| user|delete a contacts|remove contacts that I do not need to track|
```
perhaps this should be contact instead of contacts

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -2,29 +2,310 @@
+The Class Diagram below shows the overall structure of the application from the menu's point of view.
+The AllOnUs class and other classes that are coupled to the menu are therefore included.
+
+![](images/MenuFeaturesClassDiagram.png)
```
Be careful about syntaxes not included in the module, in this case (c) should be not be used (plantUML can be configured to omit this syntax), if you need help with that try the forums.

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -2,29 +2,310 @@
+### Study Manager component
+API: `StudyManager.java`
+
+![](images/ModuleClassDiagram.png)
```
1. Consider adding the visibility of the variables. 
2. Perhaps add multiplicity as well.

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -2,29 +2,310 @@
+
+Step 1: The user navigates to the Expense Tracker for the first time, and decides to set a budget of $300.
 
+![](images/BudgetingReminder0.png)
```
An object diagram may be more suitable in this context.

xseh reviewed (commented):
Good attempt! Some small changes to the UML diagram and if possible make the text more concise.

mathanmahe committed:
Fix parsing for add modules that would throw warnings for spaces within parameters

YitHien committed:
Fix bugs from PED issues

YitHien committed:
Add additional constraints on slash character positioning for add method

YitHien committed:
Apply constraints to Edit field, update UserGuide, SLAP long methods and static constants out

mathanmahe committed:
Introduce constraints for module input parameters

mathanmahe committed:
Update test cases with constraints

mathanmahe committed:
Update ModuleCalendarReader to comply with constraint

mathanmahe committed:
Update UserGuide to include improvements

mathanmahe committed:
Update JUnit tests to work with improvements

mathanmahe committed:
Add text ui tests for new changes

mathanmahe committed:
Fix build error due to flawed assertion

wli-linda committed:
Address TA code review & edit access modifiers

wli-linda committed:
Update UG and DG according to PE-D feedback

wli-linda committed:
Implement basic add contact constraint (non-empty)

wli-linda committed:
Fix AY2122S2-CS2113-F10-4/tp#120

wli-linda committed:
Merge pull request #129 from mathanmahe/modules-v2.1-enhancements

Modules v2.1 enhancements

YitHien committed:
Merge pull request #130 from YitHien/YitHien-PEDEnhancements

Expense Tracker - Fix bugs from PED

wli-linda committed:
Merge pull request #131 from wli-linda/master

Enhance Contacts Manager code quality and features

YitHien committed:
Draft up PPP

mathanmahe committed:
Make StudyManager more OOP by extracting ModuleParser class

OzairHasan committed:
Update code quality, debug and enhance performance for AllOnUs.java and StorageFile.java

mathanmahe committed:
Remove unused comments

mathanmahe committed:
Make sample add command to be consistent with constraints

mathanmahe committed:
Perform SLAP for ModuleCalendarReader class

mathanmahe committed:
Add javadoc comments for public functions

mathanmahe committed:
Match gradle checkStyle

OzairHasan committed:
Update implementation of direct manager/tracker access to use enumerations

YitHien committed:
Enhanced Developer Guide and Text UI test cases

OzairHasan committed:
Update AllOnUs.java to comply with gradle checks

YitHien committed:
Add line breaks for string outputs for consistency

OzairHasan committed:
Change EXPECTED.TXT to reflect updated output

OzairHasan committed:
Update PPP

OzairHasan committed:
Update PPP

OzairHasan committed:
Debug JUnit test iteration 1

OzairHasan committed:
Update code to pass gradle checkstyle

OzairHasan committed:
Debug JUnit test iteration 2

OzairHasan committed:
Debug JUnit test iteration 3

OzairHasan committed:
Update JUnit tests for Load Save Functionality

wli-linda committed:
Validate email for AY2122S2-CS2113-F10-4/tp#105

wli-linda committed:
Implement check for duplicate contact names

wli-linda committed:
Update behavior of duplicate name and email checks

Email validation rejects all formats considered invalid, and names aren't removed from duplicate set upon deletion of contact.

Some users might want invalid format email entries as placeholders, and duplicate name check was just having buggy behavior.

Let's,
* update the deleteContact() method to make sure the hash of a name is removed
* make sure invalid email formats only cause a reminder message to show, and doesn't affect behavior otherwise

wli-linda committed:
Catch same delimiters case

Resolves AY2122S2-CS2113-F10-4/tp#123

mathanmahe committed:
Add draft Project Portfolio

YitHien committed:
Merge pull request #132 from YitHien/YitHien-PPPDraft

PPP Draft for Yit Hien

OzairHasan committed:
Merge branch 'master' into menu-redirection-defensiveness-code-quality

wli-linda committed:
Update PPP draft

wli-linda committed:
Merge pull request #138 from wli-linda/master

Implement more PE-D and other enhancements

mathanmahe committed:
Merge branch 'master' into modules-final-updates

mathanmahe committed:
Merge pull request #141 from mathanmahe/modules-final-updates

V2.1a Enhancements for StudyManager

OzairHasan committed:
Merge branch 'master' into menu-redirection-defensiveness-code-quality

OzairHasan committed:
Resolve merge conflicts

OzairHasan committed:
Update EXPECTED.TXT

OzairHasan committed:
Update EXPECTED.TXT

OzairHasan committed:
Update EXPECTED.TXT

OzairHasan committed:
Merge pull request #135 from OzairHasan/menu-redirection-defensiveness-code-quality

Menu redirection defensiveness code quality

OzairHasan committed:
Merge pull request #137 from OzairHasan/store-load-JUnit-tests

Store and load JUnit tests

wli-linda committed:
Set theme jekyll-theme-cayman

OzairHasan committed:
Update PPP

OzairHasan committed:
Merge pull request #143 from OzairHasan/update-PPP

Update PPP

OzairHasan committed:
Update PPP link and update PPP

OzairHasan committed:
Merge pull request #144 from OzairHasan/correct-PPP-link-format

Update PPP link and update PPP

mathanmahe committed:
Change access modifiers to public and reuse StudyManager constants in ModuleParser

YitHien committed:
Fix broken links for AboutUs and UserGuide

YitHien committed:
Add sample test cases to Expense Tracker in Developer Guide

YitHien committed:
Add more instructions for manual testing in Developer Guide for Expense Tracker

YitHien committed:
Minor additions to PPP

mathanmahe committed:
Refactor boolean to comply with coding standard and change access for JUnit tests

mathanmahe committed:
Update JUnit tests for StudyManager

mathanmahe committed:
Match gradle checks

mathanmahe committed:
Rebuild after gradle error

mathanmahe committed:
Rebuild after windows build error

mathanmahe committed:
Try platform independent line separator to fix error on Windows

mathanmahe committed:
Fix minor checkStyle error

mathanmahe committed:
Use platform independent line separator to resolve error

mathanmahe committed:
Add Javadoc comments for tests

mathanmahe committed:
Update tests for Module class

mathanmahe committed:
Add Junit tests for ModuleParser

mathanmahe committed:
Add Junit tests for ModuleCalendarReader

mathanmahe committed:
Integrate UI divider with console outputs

mathanmahe committed:
Reflect the Divider integration in all test cases

mathanmahe committed:
Improve constraints of Module time parameter and find command

mathanmahe committed:
Update test cases with improvements

mathanmahe committed:
Rebuild after gradle error

mathanmahe committed:
Reattempt to build after gradle error

YitHien committed:
Minor edits to Developer Guide and User Guide

OzairHasan committed:
Update DG and UG and make small feature modification to AllOnUs.java

OzairHasan committed:
Update previous commit with files that were not staged

OzairHasan committed:
Update UG and resolve minor bug in ExpenseTracker.java

OzairHasan committed:
Resolve test errors

OzairHasan committed:
Update input.txt to reflect changes to menu goto feature

mathanmahe committed:
Try different method for time parsing compatibility issues

mathanmahe committed:
Resolve multi platform build error

mathanmahe committed:
Refactor String constants for find module validation

mathanmahe committed:
Update User Guide to be consistent with latest enhancements

YitHien committed:
Update content page of DeveloperGuide

YitHien committed:
Add Glossary section of Developer Guide

wli-linda committed:
Update handling of contacts delimiters further

mathanmahe committed:
Update DG with latest enhancements and propose new features

mathanmahe committed:
Fix dead hyperlink to PPP

mathanmahe committed:
Make regex for module parameters more strict

YitHien committed:
Minor changes to UserGuide and DeveloperGuide

YitHien committed:
Update PRs reviewed for PPP

wli-linda committed:
Update PPP and manual testing section in DG

OzairHasan committed:
Update file naming for PPP

OzairHasan committed:
Stage files left out of previous commit

mathanmahe committed:
Add minor change in documentation

YitHien committed:
Add dividers in UserGuide

YitHien committed:
Add commands into PPP

wli-linda committed:
Update UG to post-PE-D behavior of delimiters

wli-linda committed:
Update find behavior, RepoSense, and JavaDoc

YitHien committed:
Merge pull request #145 from YitHien/YitHien-DocumentationEnhancements

Expense Tracker Documentation Enhancements

OzairHasan committed:
Merge pull request #148 from OzairHasan/polish-DG-UG

Polish DG and UG for menu and save load features

wli-linda committed:
Remove printFormat method from ContactsManager

mathanmahe committed:
Merge branch 'master' into modules-tests-documentation

mathanmahe committed:
Merge pull request #150 from mathanmahe/modules-tests-documentation

Modules add tests and documentation

wli-linda committed:
Merge branch 'master' of https://github.com/wli-linda/tp into wli-linda-master

wli-linda committed:
Merge branch 'wli-linda-master'

OzairHasan committed:
Update DG, UG and UML Diagrams

mathanmahe committed:
Refactor log warnings to info

OzairHasan committed:
Update PPP

YitHien committed:
Fix bug for negative year values and printing of found expense records

OzairHasan committed:
Update PPP overview

YitHien committed:
Minor fix to UG and String constant

YitHien committed:
Fix EXPECTED.txt to pass tests

OzairHasan committed:
Update DG, UG and some code for save and load

OzairHasan committed:
Update StorageFile.java to comply with gradle check

YitHien committed:
Removed Static from some methods

wli-linda committed:
Hide contacts email format warning in load

wli-linda committed:
Update Contacts Manager UG and DG

wli-linda committed:
Improve contacts invalid command message

wli-linda committed:
Edit overall info in UG

mathanmahe committed:
Update user guide with dividers and modify FAQ question

mathanmahe committed:
Update message in edit function

mathanmahe committed:
Create appendix section in DG

mathanmahe committed:
Remove circle from class diagrams

mathanmahe committed:
Update PPP final version

mathanmahe committed:
Merge pull request #153 from mathanmahe/final-documentation-fixes

Add final documentation and bug fixes

OzairHasan committed:
Merge branch 'master' into DG-UG-updates

OzairHasan committed:
Merge pull request #154 from OzairHasan/DG-UG-updates

DG and UG updates

YitHien committed:
Merge pull request #155 from YitHien/YitHien-2.1BugFix

Fix bug for Expense Tracker and update DG

wli-linda committed:
Merge pull request #156 from wli-linda/master

Make final edits to behavior and documentation

wli-linda committed:
Merge branch 'master' into master

wli-linda committed:
Merge pull request #157 from wli-linda/master

Edit overall info in UG

OzairHasan committed:
Update UG and DG

OzairHasan committed:
Merge pull request #158 from OzairHasan/DG-UG-final

Update UG and DG

OzairHasan committed:
Update DG UG with author tags

OzairHasan committed:
Merge pull request #159 from OzairHasan/master

Update DG UG with author tags
